engine-core/src/main/java/io/nosqlbench/engine/core/lifecycle/ActivityExecutor.java

```diff
@@ -423,8 +419,6 @@ private void awaitActivityAndMotorStateAlignment() {
      * @return true, if the desired SlotState was detected
      */
     private boolean awaitMotorState(Motor m, int waitTime, int pollTime, RunState... desiredRunStates) {
-        Set<RunState> desiredStates = new HashSet<>(Arrays.asList(desiredRunStates));
```

Comment:
the code in the while loop below does not make a lot of sense (populate a map and then remove all entries?).
yet i did not see a way how this set could play a role in making it sane logic.

Reply:
It looks like I was trying to tackle a concurrency bug with some stack-local state checking, but missed some scoping (and probably made it even less clear in the process). I'm reviewing the logic here and will probably need to add some test to shore up any changes here due to the cross-cutting impact of changes on this class.

Reply:
Ah, I see what the logic was doing.
There is obviously a scoping bug, however:
The actual and desired states are both counted, then the actual states which are in fact also the desired states are removed from actual states, thus leaving the remaining states which were not desired. This simply provides a summary view which is good for diagnostics. (Also, the result of this is used for control flow)

I suspect the scoping issue was from earlier code in which the actual states were kept as a copy and mutated locally.